gnosis/safe-react-apps
# [TX BUILDER] Use transaction service for centralizing contract info retrieval

## What it solves
Resolves #169 

## How this PR fixes it
We are adding a new way to get the abis as the current one is limited to 5 requests per seconds or adding API keys from the corresponding providers.

The new way about calling:
1) [Sourcify](https://sourcify.dev/) api
2) Transaction service ([new endpoint](https://github.com/gnosis/safe-client-gateway/issues/700) through gateway)

We are going to make this calls sequentially so if the contract is not found in Sourcify then we will use the transaction service as fallback

Currently we are using a direct call to the backend services for 2) as the backend team is implementing the changes

## How to test
1) Enter a contract using TX Builder
2) Check we call sourcify ...
```
// You can get contracts published in sourcify using this request
https://sourcify.dev/server/files/contracts/{chainId}
```
3) ... and if not found then the transaction service
```
Use Safe contract for example
0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552
```
